autolink/postlink/appDelegateLinker.js

```diff
@@ -1,78 +1,145 @@
+        content = content.replace(unneededImport, "");
+        elementsRemoved++;
       }
     });
 
-    return applicationContent;
+    if (unneededImports.length === elementsRemoved) {
+      debugn("   All imports have been removed");
+    } else if (elementsRemoved === 0) {
+      warnn(
+        "   No imports could be removed. Check the manual installation docs to verify that everything is properly setup:\n   https://wix.github.io/react-native-navigation/docs/installing#native-installation"
+      );
+    } else {
+      throw new Error(
+        "Some imports were removed. Check the manual installation docs to verify that everything is properly setup:\n   https://wix.github.io/react-native-navigation/docs/installing#native-installation"
+      );
+    }
+
+    return content;
+  }
+
+  _importNavigation(content) {
+    if (!this._doesImportNavigation(content)) {
+      debugn("   Importing ReactNativeNavigation.h");
+      return content.replace(/#import\s+"AppDelegate.h"/, '#import "AppDelegate.h"\n#import <ReactNativeNavigation/ReactNativeNavigation.h>');
+    }
+
+    warnn("   AppDelegate already imports ReactNativeNavigation.h");
+    return content;
   }
 
-  _importNavigation(applicationContent) {
-    if (!this._doesImportNavigation(applicationContent)) {
-      debugn("   ");
-      return applicationContent
-        .replace(/#import\s+"AppDelegate.h"/, "#import \"AppDelegate.h\"\n#import <ReactNativeNavigation/ReactNativeNavigation.h>")
+  _bootstrapNavigation(content) {
+    if (this._doesBootstrapNavigation(content)) {
+      warnn("   Navigation Bootstrap already present.");
+      return content;
     }
-    warnn("   AppDelegate already imports Navigation");
-    return applicationContent;
+
+    debugn("   Bootstrapping Navigation");
+    return content.replace(/RCTBridge.*];/, "[ReactNativeNavigation bootstrapWithDelegate:self launchOptions:launchOptions];");
+  }
+
+  _doesBootstrapNavigation(content) {
+    return /ReactNativeNavigation\s+bootstrap/.test(content);
   }
 
-  _removeApplicationLaunchContent(applicationContent) {
-    const toRemove = [/RCTRootView\s+\*rootView((.|\r|\s)*?)];\s+/, /rootView.backgroundColor((.|\r)*)];\s+/,
-      /self.window((.|\r)*)];\s+/, /UIViewController\s\*rootViewController((.|\r)*)];\s+/, /rootViewController\.view\s+=\s+rootView;\s+/,
-      /self.window.rootViewController\s+=\s+rootViewController;\s+/, /\[self.window\s+makeKeyAndVisible];\s+/
-    ]
+  _removeApplicationLaunchContent(content) {
+    debugn("   Removing Application launch content");
+
+    const toRemove = [
+      /RCTRootView\s+\*rootView((.|\r|\s)*?)];\s+/,
+      /rootView.backgroundColor((.|\r)*)];\s+/,
+      /self.window((.|\r)*)];\s+/,
+      /UIViewController\s\*rootViewController((.|\r)*)];\s+/,
+      /rootViewController\.view\s+=\s+rootView;\s+/,
+      /self.window.rootViewController\s+=\s+rootViewController;\s+/,
+      /\[self.window\s+makeKeyAndVisible];\s+/
+    ];
+    let elementsRemoved = 0;
```

Comment:
With this naming I would expect this variable to be an array, I would name it `elementsRemovedCount` or something alike.